app/src/play/java/de/danoeh/antennapod/cast/CustomMRControllerDialog.java

```diff
@@ -0,0 +1,347 @@
+        } else {
+            if (hasTitle) {
+                titleView.setText(title);
+                showTitle = true;
+            }
+            if (hasSubtitle) {
+                subtitleView.setText(subtitle);
+                showSubtitle = true;
+            }
+        }
+        if (showSubtitle) {
+            titleView.setSingleLine();
+        } else {
+            titleView.setMaxLines(2);
+        }
+        titleView.setVisibility(showTitle ? View.VISIBLE : View.GONE);
+        subtitleView.setVisibility(showSubtitle ? View.VISIBLE : View.GONE);
+
+        updateState();
+
+        if(rootView.getVisibility() != View.VISIBLE) {
+            artView.setVisibility(View.GONE);
+            rootView.setVisibility(View.VISIBLE);
+        }
+
+        if (fetchArtTask != null) {
+            fetchArtTask.cancel(true);
+        }
+
+        fetchArtTask = new FetchArtTask(description);
+        fetchArtTask.execute();
+    }
+
+    private void updateState() {
+        if (mediaController == null) {
+            return;
+        }
+        PlaybackStateCompat state = mediaController.getPlaybackState();
+        if (state != null) {
+            boolean isPlaying = state.getState() == PlaybackStateCompat.STATE_BUFFERING
+                    || state.getState() == PlaybackStateCompat.STATE_PLAYING;
+            boolean supportsPlay = (state.getActions() & (PlaybackStateCompat.ACTION_PLAY
+                    | PlaybackStateCompat.ACTION_PLAY_PAUSE)) != 0;
+            boolean supportsPause = (state.getActions() & (PlaybackStateCompat.ACTION_PAUSE
+                    | PlaybackStateCompat.ACTION_PLAY_PAUSE)) != 0;
+            if (isPlaying && supportsPause) {
+                playPauseButton.setVisibility(View.VISIBLE);
+                playPauseButton.setImageResource(getThemeResource(getContext(),
+                        android.support.v7.mediarouter.R.attr.mediaRoutePauseDrawable));
+                playPauseButton.setContentDescription(getContext().getResources()
+                        .getText(android.support.v7.mediarouter.R.string.mr_controller_pause));
+            } else if (!isPlaying && supportsPlay) {
+                playPauseButton.setVisibility(View.VISIBLE);
+                playPauseButton.setImageResource(getThemeResource(getContext(),
+                        android.support.v7.mediarouter.R.attr.mediaRoutePlayDrawable));
+                playPauseButton.setContentDescription(getContext().getResources()
+                        .getText(android.support.v7.mediarouter.R.string.mr_controller_play));
+            } else {
+                playPauseButton.setVisibility(View.GONE);
+            }
+        }
+    }
+
+    private static int getThemeResource(Context context, int attr) {
+        TypedValue value = new TypedValue();
+        return context.getTheme().resolveAttribute(attr, value, true) ? value.resourceId : 0;
+    }
+
+    private class FetchArtTask extends AsyncTask<Void, Void, Bitmap> {
```

Comment:
Cool! I'll then change the implementation either later today or tomorrow.
I wanted a quick and simple way to access the main thread after running something asynchronously, and AsyncTask was the first that came to mind, and was also being applied in the support library for the super class.